### PR TITLE
scripts to create named workspace + folder idempotently

### PR DESCRIPTION
**Pull request by alsuren**
scripts for populating folders as part of [AB#2134](https://dev.azure.com/futurenhs/75407963-f812-43e5-a734-1059bbc036a3/_workitems/edit/2134) 

## Testing information

`./workspace-service/scripts/create-workspace-if-needed.sh dev-$FNHSNAME "Workspace by Bash" "Workspace that was made by a Bash script"`
-> should print the uuid of the workspace
-> running it again should print the same uuid

`./workspace-service/scripts/create-folder-if-needed.sh dev-$FNHSNAME "Workspace by Bash" "Folder by Bash" "Folder that was made by a Bash script"`
-> should print the uuid of the folder
-> running it again should print the same uuid

## Test:

- [x] Tested locally

### CODE REVIEW

File: workspace-service/scripts/create-folder-if-needed.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+CURRENT_CONTEXT=$(kubectl config current-context)
+if [ "$ENVIRONMENT" != "$CURRENT_CONTEXT" ]; then
+	echo "You want to populate:    $ENVIRONMENT"
+	echo "Your current content is: $CURRENT_CONTEXT"
+	echo "Please change your current context using:"
+	echo "    kubectl config use-context $ENVIRONMENT"
+	echo "or"
+	echo "    az account set --subscription \$SUBSCRIPTION_ID && az aks get-credentials --resource-group=platform-$ENVIRONMENT --name=$ENVIRONMENT"
+	echo "to chance context. Once that is done, please run:"
```
Comment (frigus02) [inline workspace-service/scripts/create-folder-if-needed.sh:26]:
```suggestion
	echo "Once that is done, please run:"
```

File: workspace-service/scripts/create-folder-if-needed.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	echo "Something went wrong finding/creating your workspace"
+	exit 1
+fi
+
+body=$(
+	jq \
+		--null-input \
+		--arg workspace "$workspace" \
+		'{
+      "query": "query FoldersByWorkspace($workspace: ID!) { foldersByWorkspace(workspace: $workspace) { title, id } }",
```
Comment (frigus02) [inline workspace-service/scripts/create-folder-if-needed.sh:43]: I think the query and mutation names are optional. They're useful when we generate types or SDKs from them. Here we could in theory leave them out, which might make it slightly easier to read? Not sure, though. Maybe better to leave as is.

File: workspace-service/scripts/create-folder-if-needed.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+)
+
+found=$(
+	echo "$existing_folders" |
+		jq \
+			-r \
+			--arg title "$FOLDER_TITLE" \
+			'.data.foldersByWorkspace | map(select(.title == $title))[0].id'
+)
+if [ "$found" != "" ] && [ "$found" != "null" ]; then
```
Comment (frigus02) [inline workspace-service/scripts/create-folder-if-needed.sh:63]: Interesting. When do we find a `null` id?

File: workspace-service/scripts/create-workspace-if-needed.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+set -euo pipefail
+
+cd $(dirname $0)
+
+USAGE="
+
+USAGE: $(basename $0) dev-\$FNHSNAME WORKSPACE_TITLE [WORKSPACE_DESCRIPTION]
+
+"
+ENVIRONMENT="${1:?"Please specify your environment name as the first parameter, e.g. dev-jane"}"
```
Comment (frigus02) [inline workspace-service/scripts/create-workspace-if-needed.sh:12]:
```suggestion
ENVIRONMENT="${1:?"${USAGE}Please specify your environment name as the first parameter, e.g. dev-jane"}"
```

File: workspace-service/scripts/create-workspace-if-needed.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+CURRENT_CONTEXT=$(kubectl config current-context)
+if [ "$ENVIRONMENT" != "$CURRENT_CONTEXT" ]; then
+	echo "You want to populate:    $ENVIRONMENT"
+	echo "Your current content is: $CURRENT_CONTEXT"
+	echo "Please change your current context using:"
+	echo "    kubectl config use-context $ENVIRONMENT"
+	echo "or"
+	echo "    az account set --subscription \$SUBSCRIPTION_ID && az aks get-credentials --resource-group=platform-$ENVIRONMENT --name=$ENVIRONMENT"
+	echo "to chance context. Once that is done, please run:"
```
Comment (frigus02) [inline workspace-service/scripts/create-workspace-if-needed.sh:24]:
```suggestion
	echo "Once that is done, please run:"
```

**Review by frigus02 — APPROVED** [review]
Looks fine to me. Left a few comments about formatting 🙂

File: workspace-service/scripts/create-folder-if-needed.sh (hunk 3 above)
Comment (alsuren) [reply workspace-service/scripts/create-folder-if-needed.sh:63]: jq's default highlighting makes it look like it's returning nothing, but it's actually returning `null`. Pipe it through `cat` to see it:
```
echo '{"data":{"foldersByWorkspace":[{"title":"Magicalness","id":"21d7f4e6-4e33-4b85-9354-92a2289bf044"},{"title":"Magicalness","id":"2922c506-4d32-4a19-8fe2-565945fb8f8c"}]}}' | jq -r --arg title 'Befold the King of Folders' '.data.foldersByWorkspace | map(select(.title == $title))[0].id' | cat
```
I guess we don't need the empty string check? Let me have a play.

**Review by frigus02 — APPROVED** [review]

**Pull request merged**